app/src/main/java/com/findmyrecycling/MainActivity.kt

```diff
@@ -22,91 +19,186 @@ import com.firebase.ui.auth.AuthUI
-            modifier = Modifier
-                .fillMaxWidth()
-                .padding(start = 4.dp, end = 4.dp, top = 2.dp, bottom = 2.dp)
-                .border(width = 1.dp,
+    var inProductName = "No item selected"
+    var selectedProduct = Product("", 0)
+
+    @Composable
+    fun RecycleSearch(products: List<Product> = ArrayList<Product>()) {
+        var recyclable by remember { mutableStateOf("") }
+        var location by remember { mutableStateOf("") }
+
+        Column {
+            TextFieldWithDropdownUsage(productsIn = products, stringResource(R.string.recyclable))
+
+
+            OutlinedTextField(
+                value = location,
+                onValueChange = { location = it },
+                label = { Text(stringResource(R.string.location)) },
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .padding(start = 4.dp, end = 4.dp, top = 2.dp, bottom = 2.dp)
+                    .border(
+                        width = 1.dp,
                         color = Color.Black,
-                        shape = RoundedCornerShape(4.dp)),
-        )
+                        shape = RoundedCornerShape(4.dp)
+                    ),
+            )
+
+
+        }
+    }
+
+    @Preview(showBackground = true)
+    @Composable
+    fun DefaultPreview() {
+        FindMyRecyclingTheme {
+            RecycleSearch()
+        }
+
+        Button(
+            onClick = {
+                signIn()
+            }
+        ) {
+            Text(text = "Logon")
+        }
+    }
 
+    fun signIn() {
+        val providers = arrayListOf(AuthUI.IdpConfig.EmailBuilder().build())
+        val signinIntent = AuthUI.getInstance()
+            .createSignInIntentBuilder()
+            .setAvailableProviders(providers)
+            .build()
 
+        signInLauncher.launch(signinIntent)
     }
-}
 
-@Preview(showBackground = true)
-@Composable
-fun DefaultPreview() {
-    FindMyRecyclingTheme {
-        RecycleSearch("Android")
+    private val signInLauncher = registerForActivityResult(
+        FirebaseAuthUIActivityResultContract()
+    ) { res ->
+        this.signInResult(res)
     }
 
-    Button(
-        onClick = {
-            signIn()
+
+    private fun signInResult(result: FirebaseAuthUIAuthenticationResult) {
+        val response = result.idpResponse
+        if (result.resultCode == ComponentActivity.RESULT_OK) {
+            var user = FirebaseAuth.getInstance().currentUser
+        }
+    }
+
+    @Composable
+    fun TextFieldWithDropdownUsage(productsIn: List<Product>, label: String = "", take: Int= 3 ) {
+
+        val dropDownOptions = remember { mutableStateOf(listOf<Product>()) }
+        val textFieldValue = remember { mutableStateOf(TextFieldValue()) }
+        val dropDownExpanded = remember { mutableStateOf(false) }
+
+        fun onDropdownDismissRequest() {
+            dropDownExpanded.value = false
+        }
+
```

Comment:
Added Autocomplete for fewer keystrokes and correcting user query and spelling errors.
https://blog.expertrec.com/autocomplete/